src/error.rs

```diff
@@ -0,0 +1,49 @@
+use goblin::error::Error as GoblinError;
+use pdb_crate::Error as PDBError;
+pub type CompactSymbolResult<T> = Result<T, CompactSymbolError>;
+
+#[derive(Fail, Debug)]
+pub enum CompactSymbolError {
+    #[fail(display = "Strings do not match")]
+    UnmatchedString,
+
+    #[fail(display = "pdb_crate error: {}", _0)]
+    PDBError(PDBError),
+
+    #[fail(display = "Feature is not implemented")]
+    UnimplementedFeature,
+
+    #[fail(display = "IO Error")]
+    IOError,
+
+    #[fail(display = "Invalid string: {}", _0)]
+    InvalidStringError(&'static str),
+
+    #[fail(display = "Invalid result: {}", _0)]
+    InvalidResultError(&'static str),
+
+    #[fail(display = "None value error")]
+    NoneValueError,
```

Comment:
Invalid string, invalid result, and none value error are a bit too generic for my taste. Let's make a different name for each specific thing that can go wrong.

Reply:
Ah yes. NoneValueError is actually propagating `std::convert::From<std::option::NoneError>`, but I cannot directly propagate that because of `use of unstable library feature 'try_trait`. I tried to pass on different headers like `unstable` but still couldn't get it to work. So I decide to just have `NoneValueError` there. 

For `InvalidStringError`, I will collapse it to `InvalidInputError` and pass on an error message describing what parameter cannot be parsed specifically so to not make it generic.